testsuite/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/WildflyHelmChartExistingValuesOpenShiftExampleApplication.java

```diff
@@ -55,20 +54,17 @@ public Map<String, String> getSetOverrides() {
 		return setOverrides;
 	}
 
-	private HelmWildflyRelease loadRelease() {
-		URL url = this.getClass().getResource("wildfly-helm-values.yaml");
-		if (url == null) {
-			throw new IllegalStateException("No wildfly-helm-values.yaml found");
-		}
-		try {
-			Path valuePath = Path.of(url.toURI());
-			ObjectMapper mapper = new ObjectMapper(new YAMLFactory().disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER));
-			return mapper.readValue(valuePath.toFile(), HelmWildflyRelease.class);
-		} catch (Error | RuntimeException e) {
-			throw e;
-		} catch (Exception e) {
-			throw new RuntimeException(e);
-		}
+	private HelmChartRelease loadRelease() {
+		if (IntersmashTestsuiteProperties.isCommunityTestExecutionProfileEnabled()) {
+			return new WildFlyHelmChartReleaseAdapter(
+					HelmChartReleaseAdapter.<HelmWildflyRelease> fromValuesFile(
+							this.getClass().getResource("wildfly-helm-values.yaml"), HelmWildflyRelease.class));
+		} else if (IntersmashTestsuiteProperties.isProductizedTestExecutionProfileEnabled()) {
+			return new Eap8HelmChartReleaseAdapter(
```

Comment:
How can we fit channels info into a release loaded from a file? perhaps we need to use `Eap8HelmChartReleaseAdapter.withBuildEnvironmentVariables` ?

Reply:
IIUC channels should be defined in the static file as well, or maybe I am not getting the point.
That being said though, after loading from a file, we could still manipulate the Helm release POJO dynamically in two different ways:
1. by using the schema POJO API, i.e. `HelmEap8Release`, since `HelmWildflyRelease` doesn't have one as it's not defined by its JSON schema.
2. by using the `WildflyHelmChartRelease`interface API, which both the `WildFlyHelmChartReleaseAdapter` and `Eap8HelmChartReleaseAdapter` implement, and specifically the following ones:
```
...
void setS2iChannels(LinkedHashSet<String> s2iFeaturePacks);

	WildflyHelmChartRelease withS2iChannels(LinkedHashSet<String> s2iChannels);

	WildflyHelmChartRelease withS2iChannel(String s2iChannel);
...
```

Reply:
I see `withS2iChannels` translates to in the EAP template, which in turn relies on variable GALLEON_PROVISION_CHANNELS to do its job:
```
      {{- if .Values.build.s2i.channels }}
      - name: GALLEON_PROVISION_CHANNELS
        value: {{ join "," .Values.build.s2i.channels | quote }}
      {{- end }}
```
any idea where `GALLEON_PROVISION_CHANNELS` is documented?

Reply:
Yep, that ENV VAR is documented for EAP 8 Beta, see the product [docs](https://access.redhat.com/documentation/it-it/red_hat_jboss_enterprise_application_platform/8-beta/html/using_jboss_eap_on_openshift_container_platform/assembly_capability-trimming-in-jboss-eap-for-openshift_default).